lib/asset-sprayer/src/lib.rs

```diff
@@ -95,6 +95,33 @@ impl AssetSprayer {
             .ok_or(AssetSprayerError::EmptyChoice)?;
         Ok(text)
     }
+
+    pub async fn raw_prompt(
+        &self,
+        prompt: &(impl RawPromptYamlSource + std::fmt::Display),
+    ) -> Result<CreateChatCompletionRequest> {
+        Ok(serde_yaml::from_str(&self.raw_prompt_yaml(prompt).await?)?)
+    }
+
+    pub async fn raw_prompt_yaml(
+        &self,
+        prompt: &(impl RawPromptYamlSource + std::fmt::Display),
+    ) -> Result<Cow<'static, str>> {
```

Comment:
I suspect that could be a `String`, but if you want clone-on-write semantics, this might be the right choice!

Reply:
In production mode, these strings are always embedded in the executable, so there's no point copying them--the reason for Cow is mainly because of dev mode.